### DISCUSSION TITLE
[Question] Project status

### DISCUSSION THREAD

**Discussion by gavr123456789** [gircore/gir.core#476] [Q&A] [answered]
Hi, I've been following the project for a year now. I am very interested to know the current status, I see a lot of commits and checkboxes in issue generator V3, but it's still not entirely clear to me. Сan you explain exactly what is happening, and how close everything is to being able to start writing GTK 4 applications.

**Comment by badcel** [comment, marked as answer]
Unfortunately the current generator code is not in a sustainable form. It is very hard to extend it or find bugs. Thus I started reimplementing it in the [generator3](https://github.com/gircore/gir.core/tree/feature/generator3) branch. Before this there was another step which split the generator into two parts (1. Gir loading + resolving 2. Template generation) which took some time, too.

This is why there were no feature updates or why the move to GTK4 get's delayed. Adding GTK4 would create more work for the current generator so it makes no sense to start with this as the rewrite would have happened never the less. For the new generator the DBus sample is working again. For GTK there is the generation for delegates and methods missing on which @mjakeman and I are working currently.

I know it is taking quite some time since I started this project but it really is no small task and I want this project to be longtime sustainable.

If you are interested in GTK4 support feel free to join our matrix channel and start working on GTK4 support for the generator3 branch. The best way to speed things up is actually to step on board. As this project is 100% C# (including the build) there are are no hurdles to get things compiling. GTK4 generation PRs are welcome.

**Comment by badcel** [comment]
In general I wanted to thank you for your support and patience, I know you are one of the earliest supporters of this project :rocket: .
